dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/User.java

```diff
@@ -1024,27 +1003,14 @@ public boolean isAuthorized( Authorities auth )
 
     public Set<UserGroup> getManagedGroups()
     {
-        Set<UserGroup> managedGroups = new HashSet<>();
-
-        for ( UserGroup group : groups )
-        {
-            managedGroups.addAll( group.getManagedGroups() );
-        }
-
-        return managedGroups;
+        return emptyIfNull( groups ).stream()
+            .flatMap( group -> emptyIfNull( group.getManagedGroups() ).stream() )
+            .collect( Collectors.toUnmodifiableSet() );
     }
 
     public boolean hasManagedGroups()
     {
-        for ( UserGroup group : groups )
-        {
-            if ( group != null && group.getManagedGroups() != null && !group.getManagedGroups().isEmpty() )
-            {
-                return true;
-            }
-        }
-
-        return false;
+        return !getManagedGroups().isEmpty();
```

Comment:
This is correct but maybe we want to avoid building that collection. I think `Stream` with `anyMatch` should work nicely.